### PR TITLE
DLP: Added sample for deidentify data stored in cloud storage

### PR DESCRIPTION
**Pull request by jainamtrivedi-crest**
- Added sample for [Deidentify Data Stored in Cloud Storage](https://cloud.google.com/dlp/docs/deidentify-storage#code-example) and test cases

### CODE REVIEW

File: dlp/api/Snippets.Tests/DeidentifyDataStoredInCloudStorageTests.cs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+            {
+                new BigQueryInsertRow(insertId: "row1") {
+                    { "Name", "James" },
+                    { "Email", "james@gmail.com" }
+                },
+                new BigQueryInsertRow(insertId: "row2") {
+                    { "Name", "Gary" },
+                    { "Email", "gary@gmail.com" }
+                }
+            };
```
Comment (annadata) [inline dlp/api/Snippets.Tests/DeidentifyDataStoredInCloudStorageTests.cs:60]: Copy paste?
The is sample uses BQ as an output to write the details of transformations done.

File: dlp/api/Snippets/DeidentifyDataStoredInCloudStorage.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                {
+                    new InfoType { Name = "PERSON_NAME" },
+                    new InfoType { Name = "EMAIL_ADDRESS" }
+                }
+            },
+            IncludeQuote = true
+        };
+
+        // Construct the actions to take after the inspection portion of the job is completed.
+        // Outputs will be stored in specified GCS output path and specified table.
```
Comment (annadata) [inline dlp/api/Snippets/DeidentifyDataStoredInCloudStorage.cs:63]: The de-identified files will be written to the the GCS bucket path specified in gcsOutputPath
The details of the transformations performed will be written to BQ table specified (datasetId, tableId).
We should document this clearly. Please update comments.

File: dlp/api/Snippets.Tests/DeidentifyDataStoredInCloudStorageTests.cs (hunk 1 above)
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets.Tests/DeidentifyDataStoredInCloudStorageTests.cs]: It was my bad. Removed the addition part.

File: dlp/api/Snippets/DeidentifyDataStoredInCloudStorage.cs (hunk 2 above)
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets/DeidentifyDataStoredInCloudStorage.cs]: Done. Thanks

File: dlp/api/Snippets.Tests/resources/test.txt [hunk 3]
```diff
@@ -1 +1 @@
-My name is Robert Frost
+My name is Robert Frost and email is test@gmail.com
```
Comment (annadata) [inline dlp/api/Snippets.Tests/resources/test.txt:1]: Please use an email address from: https://developers.google.com/style/examples#example-ip-addresses

File: dlp/api/Snippets.Tests/DeidentifyDataStoredInCloudStorageTests.cs [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+            BigQueryClient bigQueryClient = BigQueryClient.Create(_fixture.ProjectId);
+            var datasetId = $"dlp_test_dotnet_dataset_{random.Next()}";
+            var tableId = $"dlp_test_dotnet_table_{random.Next()}";
+            CreateDatasetTable(datasetId, tableId, bigQueryClient);
+
+            // Create templates for de-identification.
+            var dlp = DlpServiceClient.Create();
+            var unstructuredResponse = CreateUnstructuredTemplate(dlp);
+            var structuredResponse = CreateStructuredTemplate(dlp);
+            var imageResponse = CreateImageRedactionTemplate(dlp);
```
Comment (annadata) [inline dlp/api/Snippets.Tests/DeidentifyDataStoredInCloudStorageTests.cs:189]: use ***Template as variable name. These are DLP templates.

File: dlp/api/Snippets.Tests/resources/test.txt (hunk 3 above)
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets.Tests/resources/test.txt]: Changed to alex@example.com

File: dlp/api/Snippets.Tests/DeidentifyDataStoredInCloudStorageTests.cs (hunk 4 above)
Comment (jainamtrivedi-crest) [reply dlp/api/Snippets.Tests/DeidentifyDataStoredInCloudStorageTests.cs]: Done. Thanks

**Review by annadata — APPROVED** [review]

**Review by amanda-tarafa — APPROVED** [review]

**Pull request merged**